db/migrate/20170928142521_create_users.rb

```diff
@@ -0,0 +1,13 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
```

Comment:
Please add `NOT NULL` constraints to the relevant columns in this table. It's important to have validation in the model, as you do; but it's also vital to duplicate validations on the database side in case ActiveRecord's validation fails for some reason.

https://stackoverflow.com/questions/9286176/ruby-on-rails-how-do-i-add-a-not-null-constraint-to-an-existing-column-using-a